### ISSUE TITLE
discrepancy in how timezone unaware datetimes are used

### ISSUE THREAD

**Issue by jezemine** [pingswept/pysolar#42]
Was playing with this library today and found an inconsistency.  If you call get_altitude_fast with no timezone, it assumes UTC.  If you call get_altitude with no timezone, it assumes something else and produces a different result by 80 degrees.

If I were the decider I would require timezone-awareness on all datetimes so there's nothing to assume, but that's me.  If tzinfo is not set, just raise and be done. That would be a breaking change though, so you might anger some folks :)

```python
from pysolar.solar import get_altitude, get_altitude_fast
from datetime import datetime

# first be a good citizen and make a timezone aware datetime
d = datetime(2000, 1, 1, tzinfo=pytz.utc)
y = get_altitude(33, 44, d)
print(y) # -50.85138816460989

y = get_altitude_fast(33, 44, d)
print(y) # -50.94686162937928


# now be vague and try it with no timezone - big discrepancy!
# to me not specifying the timezone is a sin already but still...
d = datetime(2000, 1, 1)
y = get_altitude(33, 44, d)
print(y) # 31.734391266371397

y = get_altitude_fast(33, 44, d)
print(y) # -50.94686162937928
```

**Comment by pingswept** [comment]
That's definitely a bug! Thanks for reporting it.

If I remember right, when Pysolar was updated to work with Python 3, we switched to always using timezone-aware datetimes, just because the person who was writing the code wanted to (and I think it was a good move). But there was never any systematic check that caught the bug you just found.

Anyway, if anyone wanted to submit a patch that fixes this, it would be very welcome.

**Comment by DouglasAllen** [comment]
How do you do a patch? Let's just do the test first.

**Comment by jezemine** [comment]
first you fork the repo, fix the bug in your fork, then do a "pull request".  I started doing it once but got sidetracked.

**Comment by DouglasAllen** [comment]
Oh so that's what he means. I'm working on a dev branch myself with him. It's his call to merge stuff. I want to keep my master clean and fresh. I might want to get something without going out on the web to find it.
Thnaks.

**Comment by JacquotLeHaricot** [comment]
I would gladly give it a try ! 
Just to be sure, what would be the best way to deal with tz-unaware datetimes ?
* raise an exception (may not be backward-compatible)
* raise a warning and convert to UTC
* no warning and convert to UTC

**Comment by pingswept** [comment]
Pysolar should raise an exception in all cases where a tz-unaware datetime is used. It might break some installations if people are auto-updating, but I think this library is used mostly for scientific modeling, not life-critical medical equipment, so breaking a system to ensure that the results are valid is the right move. Feel free to email the mailing list if you have other general questions that aren't specific to this bug.

**Comment by pingswept** [comment]
Fixed in pull request #71.
